cli/ops/fs.rs

```diff
@@ -1280,30 +1271,16 @@ fn op_symlink_sync(
   #[cfg(not(unix))]
   {
     use std::os::windows::fs::{symlink_dir, symlink_file};
-
-    match args.options {
-      Some(options) => match options._type.as_ref() {
-        "file" => symlink_file(&oldpath, &newpath)?,
-        "dir" => symlink_dir(&oldpath, &newpath)?,
-        _ => return Err(ErrBox::type_error("unsupported type")),
-      },
-      None => {
-        let old_meta = std::fs::metadata(&oldpath);
-        match old_meta {
-          Ok(metadata) => {
-            if metadata.is_file() {
-              symlink_file(&oldpath, &newpath)?
-            } else if metadata.is_dir() {
-              symlink_dir(&oldpath, &newpath)?
-            }
-          }
-          Err(_) => return Err(ErrBox::type_error(
-            "you must pass a `options` argument for non-existent target path in windows"
-              .to_string(),
-          )),
-        }
+    if let Ok(metadata) = std::fs::metadata(&oldpath) {
```

Comment:
This introduces an extra syscall...

Reply:
@piscisaureus are you ok with this change?

Reply:
Not ok.
Auto detection requires the symlink target to exist which is not guaranteed.
Symlink types on windows are not an implementation detail.

Reply:
> Auto detection requires the symlink target to exist which is not guaranteed.

Hence doing the same as Go's implementation of assuming file when no target exists.

Reply:
Well I know of these examples:
- node.js
- git (I had to add that myself because auto detection didn't work)
- rust

Reply:
https://nodejs.org/api/fs.html#fs_fs_symlink_target_path_type_callback

Reply:
> https://nodejs.org/api/fs.html#fs_fs_symlink_target_path_type_callback

Fair enough, I'll consider Node a tie breaker.